pyScienceMode2/rehastim_interface.py

```diff
@@ -371,3 +367,527 @@ def get_motomed_torque(self) -> float:
+        """
+        extended_version_success = self.get_extended_version()
+        device_id_success = self.get_device_id()
+        stim_status_success = self.get_stim_status()
+        battery_status_success = self.get_battery_status()
+        main_status_success = self.get_main_status()
+
+        return (
+            extended_version_success,
+            device_id_success,
+            stim_status_success,
+            battery_status_success,
+            main_status_success,
+        )
+
+    @staticmethod
+    def channel_number_to_channel_connector(no_channel):
+        """
+        Converts the channel number to the corresponding channel and connector.
+        For example, if the user enters no_channel 3,
+        it will convert this number and interpret it as channel 2 of 4 [0,3] for the yellow connector.
+
+        Parameters
+        ----------
+        no_channel : int
+            The channel number.
+
+        Returns
+        -------
+        channel and connector
+        """
+        channels = [
+            sciencemode.lib.Smpt_Channel_Red,
+            sciencemode.lib.Smpt_Channel_Blue,
+            sciencemode.lib.Smpt_Channel_Black,
+            sciencemode.lib.Smpt_Channel_White,
+        ]
+
+        connectors = [sciencemode.lib.Smpt_Connector_Yellow, sciencemode.lib.Smpt_Connector_Green]
+
+        # Determine the connector
+        connector_idx = (no_channel - 1) // 4
+        connector = connectors[connector_idx]
+
+        # Determine the channel
+        channel = channels[(no_channel - 1) % 4]
+
+        return channel, connector
+
+    #  Low level commands
+
+    def ll_init(self):
+        """
+        Initialize the lower level of the device. The low-level is used for defining a custom shaped pulse.
+        Each stimulation pulse needs to triggered from the computer.
+        """
+        ll_init = sciencemode.ffi.new("Smpt_ll_init*")
+        ll_init.high_voltage_level = (
+            sciencemode.lib.Smpt_High_Voltage_Default
+        )  # This switches on the high voltage source
+        ll_init.packet_number = self.get_next_packet_number()
+
+        if not sciencemode.lib.smpt_send_ll_init(self.device, ll_init):
+            raise RuntimeError("Low level initialization failed.")
+        self.log(
+            "Low level initialized",
+            "Command sent to rehastim: {}".format(self.TypeRehap24(sciencemode.lib.Smpt_Cmd_Ll_Init).name),
+        )
+
+        self.get_next_packet_number()
+        self._get_last_ack()
+        self.check_ll_init_ack()
+
+    def check_ll_init_ack(self):
+        """
+        Check the low level initialization status.
+        """
+        if not sciencemode.lib.smpt_get_ll_init_ack(self.device, self.ll_init_ack):
+            raise RuntimeError("Low level initialization failed.")
+        generic_error_check(self.ll_init_ack, self.ERROR_MAP)
+
+    def start_ll_channel_config(
+        self,
+        no_channel,
+        points=None,
```

Comment:
Why points are an optional entry if you can not run the function if not provided ? And what are the types of points and no_channel ?

Reply:
points is a list of Point instances, and no_channel is an integer, fixed.